profiler/nsys_mine.py

```diff
@@ -0,0 +1,355 @@
+    plot.tight_layout()
+    plot.savefig("total_times_per_kernel.png")
+
+
+def _plot_total_time_and_counts(fv3_kernel_timings: Dict[str, List[int]]):
+    fig, ax = plot.subplots()
+    fig.set_figwidth(30)
+    fig.set_figheight(10)
+    kernel_names = []
+    kernel_total_time = []
+    kernel_count = []
+    for key, timings in fv3_kernel_timings.items():
+        kernel_names.append(key)
+        total_time = 0
+        hits = len(timings)
+        for v in timings:
+            total_time += v
+        kernel_total_time.append(total_time)
+        kernel_count.append(hits)
+
+    w = 0.3
+    plot.xticks(rotation=45, ha="right")
+    ax.bar(kernel_names, kernel_total_time, width=-w, align="edge", color="tab:blue")
+    ax_counts = ax.twinx()
+    ax_counts.bar(kernel_names, kernel_count, width=w, align="edge", color="tab:orange")
+    ax.set_xlabel("Kernels")
+    ax.set_ylabel("Total time in ns")
+    ax_counts.set_ylabel("# calls")
+    plot.tight_layout()
+    plot.savefig("total_times_counts_per_kernel.png")
+
+
+def _print_total_time_kernel_table(
+    fv3_kernel_timings: Dict[str, List[int]],
+    timestep_time_in_ms: float,
+    write_csv: Optional[bool] = False,
+):
+    kernels = []
+    for key, timings in fv3_kernel_timings.items():
+        total_time = 0
+        hits = len(timings)
+        for v in timings:
+            total_time += v
+        percent_of_total_in_ms = ((total_time / 1e6) / timestep_time_in_ms) * 100
+        kernels.append([key, total_time, percent_of_total_in_ms, hits])
+    kernels.sort(key=lambda x: x[1], reverse=True)  # type: ignore
+    print("Kernel time - sorted by cumulative time")
+    table = tabulate(
+        kernels,
+        headers=["Name", "Time", "%% of timestep", "Count"],
+        tablefmt="orgtbl",
+    )
+    print(f"{table}")
+    if write_csv:
+        with open("total_time_kernel.csv", "w") as csvfile:
+            csv.writer(csvfile).writerows(kernels)
+
+
+def _plot_median_time(fv3_kernel_timings: Dict[str, List[int]]):
+    fig, ax = plot.subplots()
+    fig.set_figwidth(30)
+    fig.set_figheight(10)
+    kernel_names = []
+    kernel_median_time = []
+    for key, timings in fv3_kernel_timings.items():
+        kernel_names.append(key)
+        kernel_median_time.append(np.median(timings))
+
+    ax.bar(kernel_names, kernel_median_time)
+    plot.xticks(rotation=45, ha="right")
+    plot.xlabel("Kernels")
+    plot.ylabel("Median in ns")
+    plot.tight_layout()
+    plot.savefig("median_time_per_kernel.png")
+
+
+def _print_median_time_kernel_table(
+    fv3_kernel_timings: Dict[str, List[int]],
+    write_csv: Optional[bool] = False,
+):
+    kernels = []
+    for key, timings in fv3_kernel_timings.items():
+        hits = len(timings)
+        median = np.median(timings)
+        kernels.append([key, median, hits])
+    kernels.sort(key=lambda x: x[1], reverse=True)
+    table = tabulate(kernels, headers=["Name", "Time", "Count"], tablefmt="orgtbl")
+    print(f"{table}")
```

Comment:
instead of printing and then writing, should we do:
```
original_stdout = sys.stdout
if write_csv:
   csvfile = open("median_time_kernel.csv", "w")
   sys.stdout = csvfile
print(f"{table}")
sys.stdout = original_stdout 
if write_csv:
    csvfile.close()
```
or is there a reason to want both?

Reply:
We could, I always feels perf report should stdout by default

Reply:
I'm fine with that